onnxruntime/test/autoep/library/kernels/data_types.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+#include "../example_plugin_ep_utils.h"
+
+/// <summary>
+/// Singleton that returns sets of OrtMLDataType instances using the public C API.
+/// Analogous to the internal utilities in include/onnxruntime/core/framework/data_types.h
+/// </summary>
+class MLDataTypes {
```

Comment:
Considering moving this to the public C++ API header (but not a singleton). Seems like all kernel-based plugin EPs would benefit from this.